src/System.Management.Automation/utils/PsUtils.cs

```diff
@@ -604,6 +604,30 @@ internal static bool IsRunningOnProcessorArchitectureARM()
 #endif
         }
 
+        /// <summary>
+        /// Get a temporary directory to use, needs to be unique to avoid collision
+        /// </summary>
+        internal static string GetTemporaryDirectory()
+        {
+            string tempDir = String.Empty;
+            string tempPath = Path.GetTempPath();
+            do
+            {
+                tempDir = Path.Combine(tempPath,System.Guid.NewGuid().ToString());
```

Comment:
@iSazonov although not discoverable, for advanced uses you can define XDG_CONFIG_HOME, XDG_CACHE_HOME, and XDG_DATA_HOME env vars which PowerShell will use (on non-Windows).  This was really only to support simple cases (like Puppet) to not put a barrier for adoption of PowerShell